zerver/tests/test_push_notifications.py

```diff
@@ -1150,12 +1180,41 @@ def test_modernize_apns_payload(self) -> None:
         self.assertEqual(
             modernize_apns_payload(
                 {'alert': 'Message from Hamlet',
-                 'message_ids': [3]}),
+                 'message_ids': [3],
+                 'badge': 0}),
             payload)
         self.assertEqual(
             modernize_apns_payload(payload),
             payload)
 
+    @mock.patch('zerver.lib.push_notifications.push_notifications_enabled', return_value = True)
+    def test_apns_badge_count(self, mock_push_notifications: mock.MagicMock) -> None:
+        user_profile = self.example_user('othello')
+        # Test APNs badge count for personal messages.
+        message_ids = [self.send_personal_message(self.sender,
+                                                  user_profile,
+                                                  'Content of message')
+                       for i in range(5)]
+        self.assertEqual(get_apns_badge_count(user_profile), 5)
+        num_messages = len(message_ids)
+        # Mark the messages as read and test whether
+        # the count decreases correctly.
+        for i, message_id in enumerate(message_ids):
+            do_update_message_flags(user_profile, get_client("website"), 'add', 'read', [message_id])
+            self.assertEqual(get_apns_badge_count(user_profile), num_messages - i - 1)
+
+        # Similarly, test APNs badge count for stream mention.
+        stream = self.subscribe(user_profile, "Denmark")
+        message_ids = [self.send_stream_message(self.sender,
+                                                stream.name,
+                                                'Hi, @**Othello, the Moor of Venice**')
+                       for i in range(5)]
+        self.assertEqual(get_apns_badge_count(user_profile), 5)
+        for i, message_id in enumerate(message_ids):
+            do_update_message_flags(user_profile, get_client("website"), 'add', 'read', [message_id])
+            self.assertEqual(get_apns_badge_count(user_profile), num_messages - i - 1)
```

Comment:
I think it'd be better to just make 2 stream mention messages in that first list, above, rather than 5 PMs, rather than having this semi-duplicated block of code.

Reply:
Sorry, I meant 3 PMs and 2 stream message mentions, not to delete the PMs; can you adjust this?.

Reply:
@timabbott done. Kindly have a look.